lib/measured/unit.rb

```diff
@@ -1,25 +1,22 @@
 class Measured::Unit
   include Comparable
 
+  attr_reader :name, :names, :conversion_amount, :conversion_unit
+
   def initialize(name, aliases: [], value: nil)
     @name = name.to_s
-    @names = ([@name] + aliases.map{|n| n.to_s }).sort
-
+    @names = ([@name] + aliases.map(&:to_s)).sort
     @conversion_amount, @conversion_unit = parse_value(value) if value
   end
 
-  attr_reader :name, :names, :conversion_amount, :conversion_unit
-
-  def name_eql?(name_to_compare, case_sensitive: false)
+  def name_eql?(name_to_compare)
     return false unless name_to_compare.present?
-    name_to_compare = name_to_compare.to_s
-    case_sensitive ? @name.eql?(name_to_compare) : case_insensitive(@name).include?(name_to_compare.downcase)
```

Comment:
Although small, this ternary is the unnecessary cost I'm describing in the PR description.